clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -146,11 +146,14 @@ private void addMetadataListener() {
         this.metadata.addListener(new Metadata.Listener() {
             @Override
             public void onMetadataUpdate(Cluster cluster) {
-                // if we encounter any unauthorized topics, raise an exception to the user
-                if (!cluster.unauthorizedTopics().isEmpty())
-                    throw new TopicAuthorizationException(new HashSet<>(cluster.unauthorizedTopics()));
 
                 if (subscriptions.hasPatternSubscription()) {
+
+                    for (String topic : cluster.unauthorizedTopics()) {
+                        if (subscriptions.getSubscribedPattern().matcher(topic).matches())
+                            throw new TopicAuthorizationException(topic);
```

Comment:
Maybe we may as well collect all the matching unauthorized topics before raising?